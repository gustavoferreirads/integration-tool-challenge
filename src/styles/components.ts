import styled from "styled-components";

export const Title = styled.div`
  font-family: ${({ theme }) => theme.fontFamily.secondary};
  font-size: ${({ theme }) => theme.fontSize.primary};
  line-height: 25px;
  text-align: center;
  color: ${({ theme }) => theme.color.primary};
  margin: 10px;
`;

export const SubTitle = styled.div`
  font-size: ${({ theme }) => theme.fontSize.tertiary};
  line-height: 20px;
  text-align: center;
  color: ${({ theme }) => theme.color.secondary};
  white-space: pre-wrap;
`;

export const Box = styled.div`
  display: flex;
  height: ${({ theme }) => theme.height.box};
  align-items: flex-start;
  justify-content: center;
`;
