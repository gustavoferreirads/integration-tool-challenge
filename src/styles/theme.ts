const theme = {
  color: {
    primary: "#374a5e",
    secondary: "#33475b",
    background: "#7C99B6",
    hover: "#b1bac7",
    border: "#dfe3eb",
    grey: "#CBD6E2",
  },
  padding: {
    md: "16px",
    lg: "30px",
  },
  margin: {
    sm: "16px",
    md: "20px",
  },
  fontFamily: {
    primary: "Noto Sans",
    secondary: "Noto Sans Medium",
  },
  fontSize: {
    primary: "18px",
    secondary: "16px",
    tertiary: "15px",
  },
  height: {
    box: "355.51px",
  },
  width: {
    box: "283.4px",
  },
};
export default theme;
