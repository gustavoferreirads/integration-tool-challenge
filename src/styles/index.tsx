import React from "react";
import { createGlobalStyle, css, ThemeProvider } from "styled-components";

import NotoSansMedium from "@/assets/fonts/noto-sans/NotoSans-Medium.ttf";
import NotoSansRegular from "@/assets/fonts/noto-sans/NotoSans-Regular.ttf";

import Background from "@/assets/icons/background.svg";
import Checked from "@/assets/icons/checked.svg";
import Uncheck from "@/assets/icons/uncheck.svg";

import theme from "@/styles/theme";

export const fontFaces = css`
  @font-face {
    font-family: ${theme.fontFamily.primary};
    src: url(${NotoSansRegular});
  }

  @font-face {
    font-family: ${theme.fontFamily.secondary};
    src: url(${NotoSansMedium});
  }
`;

const Style = createGlobalStyle`
${fontFaces}
*,
*::before,
*::after {
  box-sizing: border-box;
}

* {
  margin: 0;
}

html {
  font-family: ${theme.fontFamily.primary};
  font-size: ${theme.fontSize.secondary};
}

html,
body {
  height: 100%;
}

body {
  line-height: 1.5;
  -webkit-font-smoothing: antialiased;
  background-image: url(${Background});
  background-position: center;
  background-repeat: no-repeat;
}

img,
picture,
video,
canvas,
svg {
  display: block;
  max-width: 100%;
}

input,
button,
textarea,
select {
  font: inherit;
}

p,
h1,
h2,
h3,
h4,
h5,
h6 {
  overflow-wrap: break-word;
}

#root {
  isolation: isolate;
  height: 100%;
}

.container {
  display: block;
  position: relative;
  padding-left: 32px;
  margin-bottom: 10px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #33475b;
}

.container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.checkmark {
  position: absolute;
  top: 4px;
  left: 0;
  height: 16px;
  width: 16px;
  border-radius: 3px;
  content: url(${Uncheck});
}

.container:hover input ~ .checkmark {
  background-color: #7C99B6;
}

.container input:checked ~ .checkmark {
  content:  url(${Checked});
}

::-webkit-scrollbar {
  width: 3px;
  border-radius: 10px;
  height:   39px;

}

/* Track */
::-webkit-scrollbar-track {
  background: rgba(255, 255, 255, 0);
  border-radius: 10px;
}

/* Handle */
::-webkit-scrollbar-thumb {
  background: ${theme.color.grey};
  border-radius: 10px;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: ${theme.color.hover};
}
`;

interface Props {
  children?: React.ReactNode;
}

export const GlobalStyle: React.FC<Props> = ({ children }) => (
  <ThemeProvider theme={theme}>
    <Style />
    {children}
  </ThemeProvider>
);
