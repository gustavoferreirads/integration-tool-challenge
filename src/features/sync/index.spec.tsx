import { render } from "@testing-library/react";

import { SyncContactsSection } from "./index";
import { GlobalStyle } from "@/styles";

describe("Contacts Synchronization", () => {
  it("should render contact providers synchronization", () => {
    const { asFragment } = render(
      <GlobalStyle>
        <SyncContactsSection />
      </GlobalStyle>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
