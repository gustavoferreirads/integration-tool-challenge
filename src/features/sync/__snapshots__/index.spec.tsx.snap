// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts Synchronization should render contact providers synchronization 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf fDibLw"
  >
    <div
      class="sc-breuTD apBAU"
    >
      <img
        alt="Provider logo"
        data-testid="logo"
        src="test-file-stub"
      />
      <div
        class="sc-bczRLJ hKfxeo"
        data-testid="title"
      >
        Gmail
      </div>
      <div
        class="sc-gsnTZi gBjWcM"
        data-testid="sub-title"
      >
        These Gmail contacts will
 sync to MailChimp
      </div>
      <div
        class="sc-papXJ flpaCQ"
      >
        <div
          class="sc-kDDrLX dJjASk"
        >
          <div
            class="sc-crXcEl cuWCOu"
          >
            <img
              alt="check"
              src="test-file-stub"
            />
            <div
              class="sc-gsnTZi sc-iqcoie gBjWcM egOsdE"
            >
              All contacts
            </div>
          </div>
          <img
            alt="arrow"
            class="sc-jqUVSM iAfbKH"
            src="test-file-stub"
          />
        </div>
        <div
          class="sc-evZas fOvbAl"
        >
          <label
            class="container"
          >
            Family
            <input
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
          <label
            class="container"
          >
            Work Friends
            <input
              checked=""
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
          <label
            class="container"
          >
            Another label
            <input
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
          <label
            class="container"
          >
            Another label
            <input
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
          <label
            class="container"
          >
            Another label
            <input
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
          <label
            class="container"
          >
            Another label
            <input
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
        </div>
      </div>
    </div>
    <div
      class="sc-dkzDqf sc-ftvSup fDibLw bOdson"
    >
      <div
        class="sc-hKMtZM mZRne"
      >
        <img
          alt="sync"
          class="sc-eCYdqJ ldPOBz"
          src="test-file-stub"
        />
        <div
          class="sc-gsnTZi sc-jSMfEi sc-gKXOVf gBjWcM iGuYCg lhDFCn"
        >
          All done!
        </div>
        <div
          class="sc-gsnTZi sc-jSMfEi sc-iBkjds gBjWcM iGuYCg dorSkQ"
        >
          Sync Contacts
        </div>
      </div>
    </div>
    <div
      class="sc-breuTD apBAU"
    >
      <img
        alt="Provider logo"
        data-testid="logo"
        src="test-file-stub"
      />
      <div
        class="sc-bczRLJ hKfxeo"
        data-testid="title"
      >
        Mailchimp
      </div>
      <div
        class="sc-gsnTZi gBjWcM"
        data-testid="sub-title"
      >
        These Mailchimp contacts will
 sync to Gmail
      </div>
      <div
        class="sc-papXJ flpaCQ"
      >
        <div
          class="sc-kDDrLX dJjASk"
        >
          <div
            class="sc-crXcEl cuWCOu"
          >
            <img
              alt="check"
              src="test-file-stub"
            />
            <div
              class="sc-gsnTZi sc-iqcoie gBjWcM egOsdE"
            >
              All contacts
            </div>
          </div>
          <img
            alt="arrow"
            class="sc-jqUVSM iAfbKH"
            src="test-file-stub"
          />
        </div>
        <div
          class="sc-evZas fOvbAl"
        >
          <label
            class="container"
          >
            Family
            <input
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
          <label
            class="container"
          >
            Work Friends
            <input
              checked=""
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
          <label
            class="container"
          >
            Another label
            <input
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
          <label
            class="container"
          >
            Another label
            <input
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
          <label
            class="container"
          >
            Another label
            <input
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
          <label
            class="container"
          >
            Another label
            <input
              type="checkbox"
            />
            <span
              class="checkmark"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
